NCS.DSS.ContentEnhancer/Cosmos/Provider/DocumentDBProvider.cs

```diff
@@ -1,63 +1,45 @@
-﻿using Microsoft.Azure.Documents;
-using Microsoft.Azure.Documents.Client;
-using Microsoft.Azure.Documents.Linq;
-using NCS.DSS.ContentEnhancer.Cosmos.Client;
-using NCS.DSS.ContentEnhancer.Cosmos.Helper;
+﻿using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Cosmos.Linq;
+using NCS.DSS.ContentEnhancer.Models;
 
 namespace NCS.DSS.ContentEnhancer.Cosmos.Provider
 {
     public class DocumentDBProvider : IDocumentDBProvider
     {
-        private readonly IDocumentDBHelper _documentDbHelper;
-        private readonly IDocumentDBClient _databaseClient;
+        private readonly Container _container;
+        private readonly string _databaseId = Environment.GetEnvironmentVariable("DatabaseId");
+        private readonly string _containerId = Environment.GetEnvironmentVariable("CollectionId");
 
-        public DocumentDBProvider(IDocumentDBHelper documentDbHelper, IDocumentDBClient documentDbClient)
+        public DocumentDBProvider(CosmosClient cosmosClient)
         {
-            _documentDbHelper = documentDbHelper;
-            _databaseClient = documentDbClient;
+            _container = cosmosClient.GetContainer(_databaseId, _containerId);
         }
 
-        public async Task<List<Models.Subscriptions>> GetSubscriptionsByCustomerIdAsync(Guid? customerId, string senderTouchPointId)
+        public async Task<List<Subscriptions>> GetSubscriptionsByCustomerIdAsync(Guid? customerId, string senderTouchPointId)
         {
-            var collectionUri = _documentDbHelper.CreateDocumentCollectionUri();
-
-            var client = _databaseClient.CreateDocumentClient();
-
-            var query = client
-                ?.CreateDocumentQuery<Models.Subscriptions>(collectionUri)
-                .Where(x => x.CustomerId == customerId &&
-                            x.TouchPointId != senderTouchPointId &&
-                            x.Subscribe)
-                .AsDocumentQuery();
-
-            if (query == null)
+            if (customerId == null)
+            {
                 return null;
+            }
+
+            var query = _container.GetItemLinqQueryable<Subscriptions>(true)
+                .Where(x => x.CustomerId == customerId && x.TouchPointId != senderTouchPointId && x.Subscribe)
+                .ToFeedIterator();
 
-            var subscriptions = new List<Models.Subscriptions>();
+            List<Subscriptions> subscriptions = new List<Subscriptions>();
 
             while (query.HasMoreResults)
             {
-                var results = await query.ExecuteNextAsync<Models.Subscriptions>();
+                var results = await query.ReadNextAsync();
                 subscriptions.AddRange(results);
             }
 
             return subscriptions.Any() ? subscriptions : null;
         }
 
-        public async Task<ResourceResponse<Document>> CreateSubscriptionsAsync(Models.Subscriptions subscriptions)
+        public async Task<ItemResponse<Subscriptions>> CreateSubscriptionsAsync(Subscriptions subscriptions)
```

Comment:
Can we remove this please - CENHA doesn't support creation of subscriptions - this (I believe to be) redundant